release/models/nat/openconfig-nat.yang

```diff
@@ -0,0 +1,1008 @@
+        key "name";
+        description
+          "List of NAT instances.";
+
+        leaf name {
+          type leafref {
+            path "../config/name";
+          }
+          description
+            "Reference to NAT instance name.";
+        }
+
+        container config {
+          description
+            "Configuration parameters for the NAT instance.";
+
+          uses nat-instance-config;
+        }
+
+        container state {
+          config false;
+          description
+            "Operational state data for the NAT instance.";
+
+          uses nat-instance-config;
+          uses nat-instance-state;
+        }
+
+        uses nat-interface-top;
+        uses nat-pool-top;
+        uses nat-mapping-top;
+        uses nat-translation-table-top;
+      }
+    }
+  }
+
+  grouping nat-pool-top {
+    description
+      "Top-level grouping for NAT pools.";
+
+    container pools {
+      description
+        "NAT address pool configuration and state.";
+
+      list pool {
+        key "name";
+        description
+          "List of NAT address pools.";
+
+        leaf name {
+          type leafref {
+            path "../config/name";
+          }
+          description
+            "Reference to pool name.";
+        }
+
+        container config {
+          description
+            "Configuration parameters for the NAT pool.";
+
+          uses nat-pool-config;
+        }
+
+        container state {
+          config false;
+          description
+            "Operational state data for the NAT pool.";
+
+          uses nat-pool-config;
+          uses nat-pool-state;
+        }
+      }
+    }
+  }
+
+  grouping nat-mapping-top {
+    description
+      "Top-level grouping for static NAT mappings.";
+
+    container mappings {
+      description
+        "Static NAT mapping configuration and state.";
+
+      list mapping {
+        key "name";
+        description
+          "List of static NAT mappings.";
+
+        leaf name {
+          type leafref {
+            path "../config/name";
+          }
+          description
+            "Reference to mapping name.";
+        }
+
+        container config {
+          description
+            "Configuration parameters for the static mapping.";
+
+          uses nat-mapping-config;
+        }
+
+        container state {
+          config false;
+          description
+            "Operational state data for the static mapping.";
+
+          uses nat-mapping-config;
+          uses nat-mapping-state;
+        }
+      }
+    }
+  }
+
+  grouping nat-top {
+    description
+      "Top-level grouping for NAT configuration and state.";
+
+    container nat {
+      description
+        "Top-level container for NAT.";
+
+      container config {
+        description
+          "Global NAT configuration parameters.";
+
+        leaf enabled {
+          type boolean;
+          description
+            "Global administrative state for NAT functionality.";
```

Comment:
same comment as elsewhere -- what happens when this is disabled after some flow has already been NAT-ed and there is an active translation?

Reply:
https://github.com/openconfig/public/pull/1335#discussion_r2279750626